crates/subspace-service/src/lib.rs

```diff
@@ -516,3 +593,28 @@ pub async fn new_full(config: Configuration) -> Result<NewFull<Arc<FullClient>>,
         backend,
     })
 }
+
+/// Builds a new object suitable for chain operations.
+pub fn new_chain_ops(
+    mut config: &mut Configuration,
+) -> Result<
+    (
+        Arc<FullClient<subspace_runtime::RuntimeApi, SubspaceExecutorDispatch>>,
+        Arc<FullBackend>,
+        sc_consensus::BasicQueue<Block, sp_trie::PrefixedMemoryDB<BlakeTwo256>>,
+        TaskManager,
+    ),
+    Error,
+> {
+    config.keystore = sc_service::config::KeystoreConfig::InMemory;
```

Comment:
Why? It didn't seem to have been present before.

Reply:
Just follow the polkadot-service :D

Reply:
Well, if we add something, we need to know for sure why. Also we are not polkadot and our execution is not a parachain, so we shouldn't just copy what they do.

Reply:
With a quick look, I don't clearly see a reason behind this, perhaps more suitable for the nodes running in the browser from the comment of `InMemory`? They probably have a reason, but I'm not sure what it is, so you prefer keeping the old behavior?  Even so, I personally still think `new_chain_ops` is of some value since at least you don't have to face the specific runtime and ExecutorDispatch directly in subspace-node. The multiple runtime matters(excluding the test runtime) are handled in subspace-service.

Reply:
Why should subspace-service care about runtime at all, can't we inject it as a dependency instead? Same with chainspec. Looks like we just moved 90% of the node implementation into another crate.

Reply:
One benefit of integrating the subspace runtime in subspace-service is that the other conponents like cirrus will no longer need to import subspace-node that is mainly for parsing the CLI stuff, making the deps simpler.

Reply:
Cirrus will be imported by subspace-node. The fact that it is the other way around at the moment is a temporary artifact.

Reply:
Not in the near future. And I don't see the harm that subspace-service takes care of the runtime.

Reply:
I actually hope that in near future, specifically before we ship it by default. Also when we add test runtime, it should ideally not require any changes in service, otherwise we can just depend on `subspace-node`'s modules with the same success.

Reply:
The test runtime requires no changes in subspace-service because `new_full` is already generic.